booking/src/main/java/org/dcsa/conformance/standards/booking/BookingScenarioListBuilder.java

```diff
@@ -187,28 +188,40 @@ private BookingScenarioListBuilder thenHappyPathFrom(BookingState bookingState)
+  private BookingScenarioListBuilder thenHappyPathFrom(BookingState bookingState, ScenarioType scenarioType) {
+    return thenHappyPathFrom(bookingState, null, scenarioType);
+  }
+
+  private BookingScenarioListBuilder thenHappyPathFrom(BookingState bookingState, BookingState originalBookingState, ScenarioType scenarioType) {
     return switch (bookingState) {
       case CANCELLED, COMPLETED, DECLINED, REJECTED -> then(shipper_GetBooking(bookingState));
       case AMENDMENT_CONFIRMED -> then(
         shipper_GetBooking(CONFIRMED).then(
-          uc12_carrier_confirmBookingCompleted().thenHappyPathFrom(COMPLETED,null)));
+          uc12_carrier_confirmBookingCompleted().thenHappyPathFrom(COMPLETED)));
       case CONFIRMED -> then(
         shipper_GetBooking(bookingState).then(
-          uc12_carrier_confirmBookingCompleted().thenHappyPathFrom(COMPLETED,null)));
+          uc12_carrier_confirmBookingCompleted().thenHappyPathFrom(COMPLETED)));
       case PENDING_AMENDMENT -> then(
         shipper_GetBooking(PENDING_AMENDMENT).then(
-          uc7_shipper_submitBookingAmendment().thenHappyPathFrom(AMENDMENT_RECEIVED,PENDING_AMENDMENT)));
+          uc7_shipper_submitBookingAmendment().thenHappyPathFrom(AMENDMENT_RECEIVED,PENDING_AMENDMENT,scenarioType)));
       case AMENDMENT_RECEIVED -> then(
         shipper_GetBooking(originalBookingState,AMENDMENT_RECEIVED).then(
-          uc8a_carrier_approveBookingAmendment().thenHappyPathFrom(AMENDMENT_CONFIRMED,CONFIRMED)));
+          uc8a_carrier_approveBookingAmendment().thenHappyPathFrom(AMENDMENT_CONFIRMED,CONFIRMED,scenarioType)));
       case PENDING_UPDATE -> then(
         shipper_GetBooking(PENDING_UPDATE).then(
-          uc3_shipper_submitUpdatedBookingRequest().thenHappyPathFrom(UPDATE_RECEIVED,null)));
-      case UPDATE_RECEIVED, RECEIVED -> then(
-        shipper_GetBooking(bookingState).then(
-          uc5_carrier_confirmBookingRequest().thenHappyPathFrom(CONFIRMED,null)));
-      case START -> then(uc1_shipper_SubmitBookingRequest().thenHappyPathFrom(RECEIVED,null));
+          uc3_shipper_submitUpdatedBookingRequest().thenHappyPathFrom(UPDATE_RECEIVED, scenarioType)));
+      case UPDATE_RECEIVED -> then(
+        shipper_GetBooking(bookingState).thenEither(
+          uc5_carrier_confirmBookingRequest().thenHappyPathFrom(CONFIRMED, scenarioType)));
+      case RECEIVED -> then(
+        ScenarioType.DG.equals(scenarioType) || ScenarioType.REEFER.equals(scenarioType)?
+        shipper_GetBooking(bookingState).thenEither(
+          uc2_carrier_requestUpdateToBookingRequest().thenHappyPathFrom(PENDING_UPDATE, scenarioType),
+          uc5_carrier_confirmBookingRequest().thenHappyPathFrom(CONFIRMED, scenarioType)):
+          shipper_GetBooking(bookingState).then(
+            uc5_carrier_confirmBookingRequest().thenHappyPathFrom(CONFIRMED, scenarioType))
+        );
+      case START -> then(uc1_shipper_SubmitBookingRequest().thenHappyPathFrom(RECEIVED, scenarioType));
```

Comment:
Hard to read what is actually happening.

Also, I suspect the condition got inverted based on how scenarios are handled in eBL

Reply:
Flavia wanted to have the Update booking for DG and REEFER. Hence I have added this condition. 

✅ SupplyCSP [DG] - UC1 - GET - UC2 - GET - UC3 - GET - UC5 - GET - UC12 - GET
✅ SupplyCSP [DG] - UC1 - GET - UC5 - GET - UC12 - GET

✅ SupplyCSP [REEFER] - UC1 - GET - UC2 - GET - UC3 - GET - UC5 - GET - UC12 - GET
✅ SupplyCSP [REEFER] - UC1 - GET - UC5 - GET - UC12 - GET

Reply:
I am saying "The code is unreadable" and your answer "Flavia wanted the change". I am 100% certain Flavia did not ask you do write it in the most unreadable way you could, so I feel you are just not answering my review comment.

Reply:
I misunderstood ur statement. I got what you mean, probably I will rearrange the condition.